recipes/glfw/all/conanfile.py

```diff
@@ -63,7 +63,7 @@ def _configure_cmake(self):
             self._cmake.definitions["GLFW_INSTALL"] = True
             if self.settings.compiler == "Visual Studio":
                 self._cmake.definitions["USE_MSVC_RUNTIME_LIBRARY_DLL"] = "MD" in self.settings.compiler.runtime
-            self._cmake.definitions["GLFW_VULKAN_STATIC"] = self.options.vulkanStatic
+            self._cmake.definitions["GLFW_VULKAN_STATIC"] = "vulkan-loader" in self.options and not self.options["vulkan-loader"].shared
```

Comment:
```suggestion
            self._cmake.definitions["GLFW_VULKAN_STATIC"] = "vulkan-loader" in self.options and not self.options["vulkan-loader"].shared
```

Pretty sure you can link to shared Vulkan

Reply:
What do you mean? `GLFW_VULKAN_STATIC` means whether to link against static Vulkan instead of dynamically, or loading via volk, yes.

Reply:
There is no link at all when GLFW_VULKAN_STATIC=False.

Reply:
"There is no link at all" except if you use `vulkan-loader` as a dependency of the application. If you don't, you can load vulkan dynamically e.g. using volk.

Reply:
`vulkan-loader` is ignored at link time if `GLFW_VULKAN_STATIC OFF`.

Reply:
Why? Won't the application get a `-lvulkan` flag from the generator and link with vulkan? I don't understand what you're getting at.

Reply:
Well ok, but you need to set GLFW_VULKAN_STATIC OFF in this case, to avoid glfw loading it at runtime. So back to original review: GLFW_VULKAN_STATIC muste be set to ON if you don't want glfw to load it at runtime. It doen't matter if vulkan-loader is static or shared.

Reply:
> Well ok, but you need to set GLFW_VULKAN_STATIC OFF in this case, to avoid glfw loading it at runtime.

That is the default anyways.

> GLFW_VULKAN_STATIC muste be set to ON if you don't want glfw to load it at runtime.

Only on Macos, as that option only works there.

> It doen't matter if vulkan-loader is static or shared.

If you link with both Vulkan and Glfw dynamically but set `GLFW_VULKAN_STATIC` to ON, you'll trigger [this](https://github.com/glfw/glfw/blob/master/CMakeLists.txt#L51) error, so shared/static does matter:
```cmake
if (GLFW_VULKAN_STATIC)
    if (BUILD_SHARED_LIBS)
        # If you absolutely must do this, remove this line and add the Vulkan
        # loader static library via the CMAKE_SHARED_LINKER_FLAGS
        message(FATAL_ERROR "You are trying to link the Vulkan loader static library into the GLFW shared library")
```

To summarize: There are three cases:

1. Link with Vulkan and Glfw dynamically. Have to set `GLFW_VULKAN_STATIC` to OFF in order to not trigger this error. To do this, use `vulkan-loader` as dependency of the application with options `vulkan-loader:shared=True` and `glfw:shared=True`.
2. Link with Glfw either way, and don't link with Vulkan at all. Optionally load Vulkan at runtime e.g. using `volk`, or don't load it at all and use OpenGL. `GLFW_VULKAN_STATIC` needs to be OFF. To do this, do _not_ use `vulkan-loader` as a dependency.
3. Link with Vulkan and Glfw statically. This is only possible on Macos, where the Vulkan library is MoltenVK. Here `GLFW_VULKAN_STATIC` needs to be ON. To do this, use `vulkan-loader` as dependency of the application with options `vulkan-loader:shared=False` and `glfw:shared=False`.

The current state of the PR should automatically set `GLFW_VULKAN_STATIC` as described in these three cases. Did I make a mistake in the PR's code? Did I miss a case?

Reply:
1. You can link vulkan-loader shared into glfw static, on all platforms.
2. You can link vulkan-loader static into glfw static, only on Apple platforms (because vulkan-loader static is not allowed on othes platforms).
3. default behavior: vulkan-loader not linked, glfw dynamically loads vulkan-loader runtime.

Reply:
Maybe the light will come in https://github.com/glfw/glfw/issues/1845 if I have an answer 😉